miqa/core/rest/task.py

```diff
@@ -0,0 +1,33 @@
+from rest_framework import serializers
+from rest_framework.decorators import action
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.viewsets import GenericViewSet
+
+from miqa.core.models import Task
+from miqa.core.rest.user import UserSerializer
+from miqa.core.rest.experiment import ExperimentSerializer
+
+class TaskSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Task
+        fields = ['id', 'name', 'user', 'experiments']
+
+    user = UserSerializer()
+    experiments = ExperimentSerializer(many=True)
+
+
+class TaskViewSet(GenericViewSet):
+    queryset = Task.objects.all()
+    serializer_class = TaskSerializer
+    
+    permission_classes = [IsAuthenticated]
+
+    @action(detail=False, pagination_class=None)
+    def me(self, request):
```

Comment:
If this is the only method on the ViewSet, I don't see any reason not to make this `retrieve`. Is there some future use case where a supervisor/admin would want to see every task in the system (/tasks), but users would want to use a different endpoint (/tasks/me) to see only their own tasks?

Reply:
This is exactly what I was anticipating. This action gets the tasks for the current user and I assumed a supervisor view would want to be able to see/filter all tasks. If this use case does not exist then I can make this change